text/0812-tracked-built-ins.md

```diff
@@ -0,0 +1,349 @@
+  weakSet = new TrackedWeakSet<string, number>(['foo', 123]);
+}
+```
+
+or using `tracked()` decorator:
+
+```ts
+import { tracked } from 'tracked-built-ins';
+
+class Foo {
+  @tracked value = 123;
+
+  obj = tracked<{[string: number]}>({ foo: 123 });
+  arr = tracked<string, number>(['foo', 123]);
+  map = tracked<string, number>(new Map(['foo', 123]));
+  set = tracked<string, number>(new Set(['foo', 123]));
+  weakMap = tracked<{[string: number]}>(new WeakMap([[{}, 456]]));
+  weakSet = tracked<string, number>(new WeakSet(['foo', 123]));
+}
+```
+
+## Detailed design
+
+The necessary changes to `ember-cli` are relatively small since we only need
+to add the dependency to the `app` blueprint.
+
+Note that `addon` blueprint will not include `tracked-built-ins` due to
+unresolved question (at the time of writing this RFC) regarding how addons
+should declare dependencies like `@glimmer/component`, `@glimmer/tracking`, `tracked-built-ins` etc.
+
+This has the advantage (over including it as an implicit dependency), that
+apps that don't want to use it for some reason can opt out by
+removing the dependency from their `package.json` file.
+
+At the time of writing this RFC, `tracked-built-ins` provides
+autotracked alternatives only to the 6 data structure types
+most commonly used in real-world JavaScript today:
+
+- `Object`
+- `Array`
+- `Map`
+- `Set`
+- `WeakMap`
+- `WeakSet`
+
+However, this should not be considered final and immutable list of types,
+and we are open to adding other built-in data structures in the future
+*if and as usage demonstrates the necessity thereof*.
+
+The tracked data structures create a *shallow* copy of the original object
+in order to avoid its mutation which may lead to issues.
+
+The tracked data structures provided by `tracked-built-ins` addon are *not* deeply tracked.
+This can be demonstrated in such example:
+
+```ts
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedObject } from 'tracked-built-ins';
+
+export default class Scoreboard extends Component {
+  obj = new TrackedObject({
+    foo: {
+      bar: 'baz'
+    },
+    bar: 'baz'
+  });
+
+  @action
+  myAction() {
+    this.obj.bar = 'barBaz'; // is autotracked and triggers re-render.
+    this.obj.foo.bar = 'barBaz'; // is *not* autotracked and *does not* trigger re-render.
+  }
+
+  <template>
+    bar: {{this.obj.bar}}
+    foo.bar: {{this.obj.foo.bar}}
+
+    <button type="button" {{on "click" this.myAction}}>press me</button>
+  </template>
+}
+```
+
+There several reasons for that limitation:
+ - the correct semantics for a framework-level hook there aren't obvious.
+ - it is possible to build variants on it fairly cheaply in user-land given `tracked()` as updated here.
+
+## How we teach this
+
+The following guide should be added to the Autotracking In-Depth guide in the official guides, and the
+sections on [POJOs][guides-pojos] and [Array][guides-arrays] should be removed.
```

Comment:
I might be wrong, but the 4.11 docs do not seem to have been updated to reflect the addition of the `tracked-built-ins` addon and how to use it like this RFC suggested

Reply:
I think at this point, this RFC could/should be moved to the "Released" stage.
Making sure the docs are updated are definitely a requirement for getting it to the "Recommended" stage if I'm not mistaken.